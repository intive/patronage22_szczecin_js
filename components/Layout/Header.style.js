import styled from 'styled-components'

export const HeaderContainer = styled.header`
  position: sticky;
  left: 0;
  top: 0;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  height: 4.5rem;
  padding: 0.75rem 2.5rem;
  background: #FFFFFF;
`
export const HeaderTitle = styled.div`
  display: flex;
  align-items: center;
  font-weight: bold;
  font-size: 1.25rem;
  line-height: 1.5rem;
  `

export const HeaderTitleText = styled.p`
  margin-left: 0.688rem;
  color: #222222;
`
