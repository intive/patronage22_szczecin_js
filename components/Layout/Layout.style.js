import styled from 'styled-components'

export const Wrapper = styled.main`
  max-width: 1400px;
  width: 95%;
  margin: 0 auto;
`
