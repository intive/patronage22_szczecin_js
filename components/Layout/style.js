import styled from 'styled-components'

export const Wrapper = styled.main`
  padding: 0 5%;

  @media (min-width: 1440px) {
    padding: 0 2.5rem;
  }
`
