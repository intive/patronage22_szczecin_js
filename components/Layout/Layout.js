import { Wrapper } from './style'
import { useTranslation } from 'next-i18next'
import Header from '../Header/Header'
import BoardHeader from '../BoardHeader/BoardHeader'

export default function Layout (props) {
  const { t } = useTranslation('common')

  return (
    <>
      <Header title={t('headerTitle')} />
      <BoardHeader returnLinkText={t('boardHeader.returnLinkText')} buttonText={t('boardHeader.buttonText')} title='Board name 1' />
      <Wrapper>{props.children}</Wrapper>
    </>
  )
}
