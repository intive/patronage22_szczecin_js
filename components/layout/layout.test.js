/* globals describe, it, expect  */

import { render } from '@testing-library/react'
import Layout from './Layout'

describe('Header', () => {
  it('should render Layout', () => {
    const { container } = render(<Layout />)
    expect(container).toMatchSnapshot()
  })
})
