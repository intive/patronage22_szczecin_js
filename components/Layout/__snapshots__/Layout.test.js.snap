// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render Layout 1`] = `
<div>
  <header
    class="sc-bdvvtL fJsLPe"
    data-testid="header"
  >
    <div
      class="sc-gsDKAQ jYlgVk"
    >
      <a
        class="sc-dkPtRN ijCfmr"
        href="/"
      >
        <span
          class="material-icons"
        >
          crop_original
        </span>
        <p
          class="sc-hKwDye bcymKS"
        >
          headerTitle
        </p>
      </a>
    </div>
  </header>
</div>
`;
